.github/workflows/ci.yaml

```diff
@@ -387,24 +417,24 @@ jobs:
             ${{ env.BUILD_DIR }}vmlinux \
             ${{ env.BUILD_DIR }}${{ env.IMAGE_PATH }}
 
-          size \
+          .github/workflows/size.sh \
```

Comment:
Hi All!

I have a question regarding this part. I came upon an issue where `size` does not support reading LLVM IR bitcode, so I've added a script which filters out bitcode from object files. Is this acceptable?

Other solutions which came to my mind are:

- Disabling this reporting step for ThinLTO builds;
- Adding a separate step for ThinLTO builds where ELF object files are inspected using `size` and LLVM IR bitcode files are inspected using `llvm-bcanalyzer`. However `llvm-bcanalyzer`'s output is a bit verbose. I've attached an example below.

<details>
  <summary>llvm-bcanalyzer output for ./samples/rust/rust_minimal.mod.o</summary>

```
Summary of ./samples/rust/rust_minimal.mod.o:
         Total size: 774112b/96764.00B/24191W
        Stream type: LLVM IR
  # Toplevel Blocks: 4

Per-block Summary:
  Block ID #0 (BLOCKINFO_BLOCK):
      Num Instances: 1
         Total Size: 768b/96.00B/24W
    Percent of file: 0.0992%
      Num SubBlocks: 0
        Num Abbrevs: 18
        Num Records: 3
    Percent Abbrevs: 0.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      3        60      20.0          SETBID

  Block ID #8 (MODULE_BLOCK):
      Num Instances: 1
         Total Size: 3832b/479.00B/119W
    Percent of file: 0.4950%
      Num SubBlocks: 11
        Num Abbrevs: 3
        Num Records: 20
    Percent Abbrevs: 15.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      8      1003     125.4   12.50  GLOBALVAR
		      4       744     186.0          SECTIONNAME
		      2       306     153.0          FUNCTION
		      1       219                    HASH
		      1       226            100.00  SOURCE_FILENAME
		      1        35            100.00  VSTOFFSET
		      1       633                    DATALAYOUT
		      1       315                    TRIPLE
		      1        21                    VERSION

  Block ID #9 (PARAMATTR_BLOCK):
      Num Instances: 1
         Total Size: 85b/10.62B/2W
    Percent of file: 0.0110%
      Num SubBlocks: 0
        Num Abbrevs: 0
        Num Records: 1
    Percent Abbrevs: 0.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      1        21                    ENTRY

  Block ID #10 (PARAMATTR_GROUP_BLOCK_ID):
      Num Instances: 1
         Total Size: 3797b/474.62B/118W
    Percent of file: 0.4905%
      Num SubBlocks: 0
        Num Abbrevs: 0
        Num Records: 1
    Percent Abbrevs: 0.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      1      3735                    ENTRY

  Block ID #11 (CONSTANTS_BLOCK):
      Num Instances: 1
         Total Size: 4033b/504.12B/126W
    Percent of file: 0.5210%
      Num SubBlocks: 0
        Num Abbrevs: 4
        Num Records: 126
    Percent Abbrevs: 72.2222%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		     41       574      14.0  100.00  SETTYPE
		     36       544      15.1  100.00  INTEGER
		     28       448      16.0          NULL
		      6       626     104.3  100.00  AGGREGATE
		      5       807     161.4  100.00  CSTRING
		      4       256      64.0          CE_INBOUNDS_GEP
		      3        48      16.0          UNDEF
		      2        52      26.0  100.00  CE_CAST
		      1       464            100.00  STRING

  Block ID #13 (IDENTIFICATION_BLOCK_ID):
      Num Instances: 1
         Total Size: 214b/26.75B/6W
    Percent of file: 0.0276%
      Num SubBlocks: 0
        Num Abbrevs: 2
        Num Records: 2
    Percent Abbrevs: 100.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      1        11            100.00  EPOCH
		      1        71            100.00  STRING

  Block ID #14 (VALUE_SYMTAB):
      Num Instances: 1
         Total Size: 117b/14.62B/3W
    Percent of file: 0.0151%
      Num SubBlocks: 0
        Num Abbrevs: 1
        Num Records: 0

  Block ID #15 (METADATA_BLOCK):
      Num Instances: 1
         Total Size: 681813b/85226.62B/21306W
    Percent of file: 88.0768%
      Num SubBlocks: 0
        Num Abbrevs: 6
        Num Records: 3470
    Percent Abbrevs: 0.1729%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		   2206    326074     147.8          DERIVED_TYPE
		    472     49408     104.7          NODE
		    395     72176     182.7          COMPOSITE_TYPE
		    161      8372      52.0          SUBROUTINE_TYPE
		    107      5750      53.7          FILE
		     35      1082      30.9          VALUE
		     30      1458      48.6          ENUMERATOR
		     19      1102      58.0          SUBRANGE
		     13       958      73.7          BASIC_TYPE
		      7       448      64.0          GLOBAL_VAR_EXPR
		      7       364      52.0          GLOBAL_DECL_ATTACHMENT
		      7       946     135.1          GLOBAL_VAR
		      3       228      76.0          NAMED_NODE
		      3       334     111.3  100.00  NAME
		      1     41356            100.00  INDEX
		      1        68            100.00  INDEX_OFFSET
		      1    171136            100.00  STRINGS
		      1        22                    EXPRESSION
		      1       184                    COMPILE_UNIT

  Block ID #17 (TYPE_BLOCK_ID):
      Num Instances: 1
         Total Size: 65909b/8238.62B/2059W
    Percent of file: 8.5141%
      Num SubBlocks: 0
        Num Abbrevs: 6
        Num Records: 1119
    Percent Abbrevs: 94.8168%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		    332      4648      14.0  100.00  POINTER
		    289     33952     117.5  100.00  STRUCT_NAME
		    240     17782      74.1  100.00  STRUCT_NAMED
		    149      6819      45.8  100.00  FUNCTION
		     51      1130      22.2  100.00  ARRAY
		     49      1078      22.0          OPAQUE
		      6       150      25.0          INTEGER
		      1        16                    METADATA
		      1        16                    VOID
		      1        28                    NUMENTRY

  Block ID #20 (GLOBALVAL_SUMMARY_BLOCK):
      Num Instances: 1
         Total Size: 885b/110.62B/27W
    Percent of file: 0.1143%
      Num SubBlocks: 0
        Num Abbrevs: 6
        Num Records: 11
    Percent Abbrevs: 72.7273%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      8       368      46.0  100.00  PERMODULE_GLOBALVAR_INIT_REFS
		      1        22                    BLOCK_COUNT
		      1        22                    FLAGS
		      1        22                    VERSION

  Block ID #21 (OPERAND_BUNDLE_TAGS_BLOCK):
      Num Instances: 1
         Total Size: 853b/106.62B/26W
    Percent of file: 0.1102%
      Num SubBlocks: 0
        Num Abbrevs: 0
        Num Records: 6
    Percent Abbrevs: 0.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      6       774     129.0          OPERAND_BUNDLE_TAG

  Block ID #22 (METADATA_KIND_BLOCK):
      Num Instances: 1
         Total Size: 4885b/610.62B/152W
    Percent of file: 0.6310%
      Num SubBlocks: 0
        Num Abbrevs: 0
        Num Records: 31
    Percent Abbrevs: 0.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		     31      4815     155.3          KIND

  Block ID #23 (STRTAB_BLOCK):
      Num Instances: 1
         Total Size: 2294b/286.75B/71W
    Percent of file: 0.2963%
      Num SubBlocks: 0
        Num Abbrevs: 1
        Num Records: 1
    Percent Abbrevs: 100.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      1      2187            100.00  BLOB

  Block ID #25 (SYMTAB_BLOCK):
      Num Instances: 1
         Total Size: 4310b/538.75B/134W
    Percent of file: 0.5568%
      Num SubBlocks: 0
        Num Abbrevs: 1
        Num Records: 1
    Percent Abbrevs: 100.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      1      4203            100.00  BLOB

  Block ID #26:
      Num Instances: 1
         Total Size: 245b/30.62B/7W
    Percent of file: 0.0316%
      Num SubBlocks: 0
        Num Abbrevs: 0
        Num Records: 2
    Percent Abbrevs: 0.0000%

	Record Histogram:
		  Count    # Bits     b/Rec   % Abv  Record Kind
		      2       172      86.0          UnknownCode1
```
</details>

What do you all think?